utilities/sourceEvents.py

```diff
@@ -224,6 +227,10 @@ def parse(content, gmaps):
                         entry['location'] = {'description': pe['location']}
                         print("calendarId: %s, dataSourceEventId: %s,  location: %s geolocation not found" %
                               (entry.get('calendarId'), entry.get('dataSourceEventId'), entry.get('location')))
+            else:
+                entry['location'] = {
+                            'description': ""
+                        }
 
         entry_location = entry['location']
```

Comment:
@bingzhang, If an event is virtual (isVirtual = True) and its location is null in XML data, in line 235, there is an error, as the code tries to access the location. Please fix this. Also, please update CHANGELOG to include adding detect secrets. Thanks.

```
entry_location = entry['location']
KeyError: 'location'
There is exception 'location', hidden in url: https://xml.calendars.illinois.edu/eventXML13/7.xml
```